documentation/src/main/asciidoc/ch07.asciidoc

```diff
@@ -1,3 +1,324 @@
+
+For all these methods, you need to pass a node name: it is the name included in
+the node added to the property path of the constraint violation. As mentioned
+earlier, if the node name is `null`, no node is added to the property path:
+it is be useful for pure wrapper types similar to `Optional`.
+
+The choice of the method used is important as it adds important information to the
+property path of the constraint violation e.g. the index or the key of the
+validated value.
+
+=== Non generic containers
+
+You might have noticed that, until now, we only implemented value extractors
+for generic containers.
+
+Hibernate Validator also supports value extraction for non generic containers.
+
+Let's take the case of `java.util.OptionalInt` which wraps a primitive `int`
+into an ``Optional``-like container.
+
+A first attempt at a value extractor for `OptionalInt` would look like:
+
+[[example-valueextraction-optionalint-valueextractor]]
+.A `ValueExtractor` for `OptionalInt`
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/org/hibernate/validator/referenceguide/chapter07/nongeneric/OptionalIntValueExtractor.java[tags=include]
+----
+====
+
+There is an obvious thing missing for a non generic container: we don't have
+a type parameter. It has two consequences:
+
+ * we cannot determine the type of the validated value using the type argument;
+ * we cannot add constraints on the type argument (e.g. `Container<@NotNull String>`).
+
+First things first, we need a way to tell Hibernate Validator that the value
+extracted from an `OptionalInt` is of type `Integer`.
+As you can see in the above example, the `type` attribute of the `@ExtractedValue`
+annotation allows to provide this information to the validation engine.
+
+Then you have to tell the validation engine that the `Min` constraint you want to
+add to the `OptionalInt` property concerns the wrapped value and not the wrapper.
+
+Bean Validation provides the `Unwrapping.Unwrap` payload for this situation:
+
+[[example-valueextraction-optionalint-unwrapping]]
+.Using `Unwrapping.Unwrap` payload
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/org/hibernate/validator/referenceguide/chapter07/Container.java[tags=optionalIntUnwrap]
+----
+====
+
+If we take a step back, most - if not all - the constraints we would like to add to an
+`OptionalInt` property would concern the wrapped value so having a way to make it
+the default would be nice.
+
+This is exactly what the `@UnwrapByDefault` annotation is for:
+
+[[example-valueextraction-optionalint-unwrapbydefault-valueextractor]]
+.A `ValueExtractor` for `OptionalInt` marked with `@UnwrapByDefault`
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/org/hibernate/validator/referenceguide/chapter07/nongeneric/UnwrapByDefaultOptionalIntValueExtractor.java[tags=include]
+----
+====
+
+When declaring this value extractor for `OptionalInt`, constraint annotations will
+by default concern the wrapped value:
```

Comment:
You're using "concern" a lot, maybe you could replace some occurrences with "relate to" or "will be applied to".